src/parser/warlock/affliction/modules/talents/ShadowEmbrace.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+import Enemies from 'parser/shared/modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import getDamageBonus from 'parser/monk/brewmaster/modules/core/GetDamageBonus';
+import StatisticListBoxItem from 'interface/others/StatisticListBoxItem';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage, formatThousands } from 'common/format';
+import { encodeTargetString } from 'parser/shared/modules/EnemyInstances';
+
+const BONUS_PER_STACK = 0.03;
+const BUFFER = 50; // for some reason, changedebuffstack triggers twice on the same timestamp for each event, ignore an event if it happened < BUFFER ms after another
+const debug = false;
+
+class ShadowEmbrace extends Analyzer {
```

Comment:
Should this class have uptime recommendations? Maybe a Threshold for how long you should be sitting at 3 stacks?